<?xml version="1.0"?>
<launch>

  <arg name="map_topic" default="map" />
  <arg name="map_frame" default="map" />
  <arg name="filter" default="false" />
  <arg name="gt_classify" default="false" />
  <arg name="scan_topic" default="/velodyne_points" />
  <arg name="publish_sub_pts" default="true" />
  <arg name="init_map_path" default="" />

  <node pkg="point_slam" type="frame_update" name="frame_update" output="screen">
  
    <param name="odom_frame" value="odom"/>
    <param name="velo_frame" value="velodyne"/>
    <param name="map_topic" value="$(arg map_topic)"/>
    <param name="map_frame" value="$(arg map_frame)"/>
    
    <param name="publish_sub_pts" value="$(arg publish_sub_pts)"/>
    
    <param name="filter" value="$(arg filter)"/>
    <param name="gt_classify" value="$(arg gt_classify)"/>


    <param name="n_threads" value="8"/>

    <param name="map_voxel_size" value="0.06"/>         <!-- Originally 0.03 -->
    <param name="frame_voxel_size" value="0.13"/>       <!-- Originally 0.10 -->
    <param name="map2d_pixel_size" value="0.12"/>
    <param name="map2d_max_count" value="4"/>
    <param name="map2d_z_min" value="0.18"/>
    <param name="map2d_z_max" value="1.2"/>
    <param name="motion_distortion" value="true"/>
    <param name="lidar_n_lines" value="32"/>

    <!-- New parameter to provide the path to the map ply file -->
    <!-- <param name="init_map_path" value=""/> -->
    <param name="init_map_path" value="$(arg init_map_path)"/>
    
    <param name="h_scale" value="0.5"/>
    <param name="r_scale" value="4.0"/>
    <param name="outl_rjct_passes" value="2"/>
    <param name="outl_rjct_thresh" value="0.003"/>

    <param name="icp_samples" value="500"/>           <!-- Originally 600 -->
    <param name="icp_pairing_dist" value="2.0"/>
    <param name="icp_planar_dist" value="0.12"/>      <!-- Originally 0.12 -->
    <param name="icp_avg_steps" value="1"/>           <!-- Originally 5 -->
    <param name="icp_max_iter" value="50"/>           <!-- Originally 100 -->

  </node>

</launch>